### DISCUSSION TITLE
Multiple editors on the same page

### DISCUSSION THREAD

**Discussion by aleckravets** [microsoft/monaco-editor#3816] [Q&A]
What is the best way to build a notebook editor app where you have multiple cell editors on the same screen. Like in this VS Code extension - [.NET Interactive Notebooks](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode).

The number of cells in the notebook may be up to several hundreds, so If I create editors for all cells at the same time it will kill the performance. 
Is there a way to show a lightweight read-only version and only create actual editor on focusing the cell? How it's done in the mentioned VS Code extension?

![image](https://user-images.githubusercontent.com/5873219/154478063-1c0119b7-a884-4794-9b6e-47e70e8057ff.png)

Any help is appreciated

**Comment by hediet** [comment]
As long as you don't show all thousands of editors at the same time, you can just reuse those monaco editors that got scrolled out of the view, but keep all the models.
Afaik this is how notebooks are implemented, but @rebornix knows more.

**Comment by Pfoucht** [comment]
> As long as you don't show all thousands of editors at the same time, you can just reuse those monaco editors that got scrolled out of the view, but keep all the models. Afaik this is how notebooks are implemented, but @rebornix knows more.

@hediet Is this doable in a scroll-based notebooks app?

**Comment by SimonCockx** [comment]
@aleckravets Did you manage to do this? I'm in a similar situation, where I need to have multiple "files" open at once.
